tests/conjur_variable/test_cases/retrieve-variable-into-file/tests/test_default.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+import os
+import testinfra.utils.ansible_runner
+
+testinfra_hosts = [os.environ['COMPOSE_PROJECT_NAME'] + '_ansible_1']
+
+
+def test_retrieved_secret(host):
+    secret_path_file = host.file('/conjur_secret_path.txt')
+    assert secret_path_file.exists
+
+    secret_path = host.check_output("cat /conjur_secret_path.txt", shell=True)
+    secret_file = host.file(secret_path)
+    assert secret_file.exists
+    assert secret_file.mode == 0o600
+
+    secret = host.check_output("cat {0}".format(secret_path), shell=True)
```

Comment:
@jcosteatcyberark, just for my own understanding... Can you explain what this line is doing? I know it's doing a level of indirection (taking `secret_path` and `cat`ing the file that it points to), but I'd like to understand better.

Reply:
Hi @diverdane, the test playbook stores the lookup output in  `/conjur_secret_path.txt`. This file contains the path to the file that contains the secret (`secret_path`).
To resume this test asserts that `cat (cat /conjur_secret_path) = secret`

Reply:
since it wasn't clear to Dane from the code, is it worth writing this in a more-verbose-but-more-clear way or at least adding a code comment?

Reply:
Hi @izgeri I cleaned up the code and commented. It should be easier to understand now.